src/clr-angular/progress/progress-bars/progress-bar.ts

```diff
@@ -0,0 +1,116 @@
+ */
+import { Component, Input, HostBinding, Inject } from '@angular/core';
+import { UNIQUE_ID, UNIQUE_ID_PROVIDER } from '../../utils/id-generator/id-generator.service';
+import { isBooleanAttributeSet } from '../../utils/component/is-boolean-attribute-set';
+
+const PROGRESSBAR_BASE_CLASS = 'progress';
+const PROGRESSBAR_CLASSES = {
+  // Style
+  labeled: 'labeled',
+  loop: 'loop',
+  // Colors
+  success: 'success',
+  danger: 'danger',
+  // Animations
+  fade: 'progress-fade',
+  flash: 'flash',
+  flashDanger: 'flash-danger',
+};
+
+@Component({
+  selector: 'clr-progress-bar',
+  providers: [UNIQUE_ID_PROVIDER],
+  template: `
+    <progress [id]="id" [attr.max]="max" [attr.value]="value" [attr.data-displayval]="displayValue"></progress>
+    <span *ngIf="displayAriaLive()" [attr.aria-live]="ariaLive">{{ displayValue }}</span>
+  `,
+})
+export class ClrProgressBar {
+  /**
+   * Handle component ID
+   */
+  private _ID: string;
+  @HostBinding('attr.id') externalId: string = '';
+  @Input()
+  set id(value: string) {
+    this._ID = value;
+    this.externalId = null;
+  }
+  get id() {
+    return this._ID;
+  }
+
+  // Progress
+  @Input('clrMax') max: number = 100;
+  @Input('clrValue') value: number = 0;
+  @Input('clrDisplayval') displayval: string;
+
+  // Styles
+  @Input('clrLabeled') labeled: boolean;
+  @Input('clrFade') fade: boolean;
+  @Input('clrLoop') loop: boolean;
+  @Input('clrSuccess') success: boolean;
+  @Input('clrDanger') danger: boolean;
+  @Input('clrFlash') flash: boolean;
+  @Input('clrFlashDanger') flashDanger: boolean;
+
+  // Aria Live
+  @Input('clrAssertive') assertive: boolean;
+  @Input('clrOff') off: boolean;
+
+  @Input('class') userClasses: string = '';
+
+  @HostBinding('class')
+  get elementClass() {
+    const classes = [PROGRESSBAR_BASE_CLASS];
+
+    ['labeled', 'fade', 'loop', 'success', 'danger', 'flash', 'flashDanger'].forEach(cls => {
+      if (isBooleanAttributeSet(this[cls])) {
+        classes.push(PROGRESSBAR_CLASSES[cls]);
+      }
+    });
+
+    // Any additional class that the user want to add
+    if (this.userClasses.length) {
+      classes.push(this.userClasses);
+    }
+
+    return classes.join(' ');
+  }
+
+  /**
+   * Make sure that we always will have something that is readable
+   * for the screen reader
+   */
+  get displayValue() {
+    if (this.displayval) {
+      return this.displayval;
+    }
+    return `${this.value}%`;
+  }
+
+  /**
+   * Display aria-live only when there is value and it's not 0 or equal to the max value
+   */
+  displayAriaLive() {
+    return (this.value !== undefined || this.value !== 0) && this.value !== this.max;
+  }
+
+  get ariaLive() {
+    if (isBooleanAttributeSet(this.assertive)) {
+      return 'assertive';
+    }
+    if (isBooleanAttributeSet(this.off)) {
+      return 'off';
+    }
+    return 'polite';
+  }
+
+  constructor(@Inject(UNIQUE_ID) progressBarId: string) {
+    this.id = progressBarId;
```

Comment:
I don't understand the purpose of this, a random ID doesn't give the application a way to know what ID will be assigned, and there is no use of this except to set the ID. Am I missing something or can we get away without setting a unique ID, and let apps set their own ID if they wish?

Reply:
https://github.com/vmware/clarity/pull/3533/files/35efa317d0609fde3951ceb33b2d77d44e9710d5#r299520849

Did you mean something like `${progressBarId}progressbar`

Reply:
That commit doesn't show up, but I mean why do we need to set a default ID? If the user sets one, we pass it along otherwise can it just be empty?